db/db_impl/db_impl_open.cc

```diff
@@ -224,6 +224,12 @@ Status DBImpl::ValidateOptions(
     if (!s.ok()) {
       return s;
     }
+    if (cfd.name == kDefaultColumnFamilyName) {
+      if (cfd.options.disallow_memtable_writes) {
```

Comment:
I'm wondering if this is too restrictive, as it forces creation of a new CF for ingestion only. We won't be able to fail writes to a WriteBatch, but it'll fail in Write(). Its no different from someone trying to write a WriteBatch to a read-only db.

Reply:
It's a judgment call I don't feel strongly about. I'm OK with opting for somewhat inconsistent disallow behavior to support disallowing on default CF. That works for me.

Reply:
Actually, the real reason I avoided it is because rejecting at memtable write time could currently make the DB unrecoverable without manual intervention, because of a rejected write batch in WAL. That's OK if the reason you see the error is because you changed a disallow_memtable_writes setting, but not OK if you accidentally attempted a disallowed write. https://github.com/facebook/rocksdb/issues/13429

Reply:
Could we have a flag in WriteBatch to indicate whether it contains writes to the default CF? Then DBImpl::WriteImpl() can check and reject before attempting to write to the memtable.